packages/tools/src/store/ToolGroupManager/ToolGroup.ts

```diff
@@ -669,6 +682,38 @@ export default class ToolGroup implements IToolGroup {
     return cloneDeep(_configuration);
   }
 
+  public clone({ newToolGroupId, fnToolFilter = null }) {
+    let toolGroup = ToolGroupManager.getToolGroup(newToolGroupId);
```

Comment:
isn't `this` the toolGroup?

Reply:
That is why I renamed it to `newToolGroupId` because it is checking if the new `ToolGroup` that is being created exists.

```javascript
firstToolGroup.clone({ newToolGroupId: 'secondToolGroup' });
```

In that case `this` would be `firstToolGroup` but it is checking if there is a `"secondToolGroup"` tool group. When that parameter was called `toolGroup` it was confusing because people could think it was the id of the current tool group (the one that was being cloned).